pkg/chunk/purger/blocks_purger_api.go

```diff
@@ -0,0 +1,116 @@
+package purger
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/thanos-io/thanos/pkg/objstore"
+
+	cortex_tsdb "github.com/cortexproject/cortex/pkg/storage/tsdb"
+	"github.com/cortexproject/cortex/pkg/tenant"
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+type BlocksPurgerAPI struct {
+	bucketClient objstore.Bucket
+}
+
+func NewBlocksPurgerAPI(storageCfg cortex_tsdb.BlocksStorageConfig, logger log.Logger, reg prometheus.Registerer) (*BlocksPurgerAPI, error) {
+	bucketClient, err := createBucketClient(storageCfg, logger, reg)
+	if err != nil {
+		return nil, err
+	}
+
+	return newBlocksPurgerAPI(bucketClient), nil
+}
+
+func newBlocksPurgerAPI(bkt objstore.Bucket) *BlocksPurgerAPI {
+	return &BlocksPurgerAPI{bucketClient: bkt}
+}
+
+func (api *BlocksPurgerAPI) DeleteTenant(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	userID, err := tenant.TenantID(ctx)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	err = cortex_tsdb.WriteTenantDeletionMark(r.Context(), api.bucketClient, userID)
```

Comment:
Should we check if already exists, in order to not overwrite it? May be useful to not overwrite, to make sure that the "deletion time" in the marker is the oldest one.

Reply:
It doesn't make difference from API point of view, and overwriting with more recent timestamp will not do any harm. Additionally, it would be best-effort-attempt only. I suggest that we instead log a message on successful creation, so that it is possible to find out if there were multiple markers, if needed for troubleshooting.

Reply:
It's true doesn't cause any harm and it's true would be a best-effort attempt, but I still believe keeping the original deletion timestamp in the marker would be useful for debugging (eg. you're assuming every Cortex users collects and stores Cortex logs for a reasonable amount of time, which I'm not sure is true).

Reversing your answer, I may say that even checking for existence and returning a "no content" status code without overwriting it doesn't cause any harm and would avoid to overwrite it under normal circumnstances.

In any case, not a blocker to me.